src/env/commands/fossil.S

```diff
@@ -75,18 +66,19 @@ no_dosemu:
 	
         # check if we are already installed.
 dosemu_found:
-	movb    $DOSEMU_HELPER_SERIAL,%al
-	movb    $DOSEMU_HELPER_SERIAL_CHECK,%ah
-	int     $DOSEMU_HELPER_INT
+	movb    $DOS_HELPER_SERIAL_HELPER,%al
+	movb    $DOS_SUBHELPER_SERIAL_TSR_CHECK,%ah
```

Comment:
Nope
commit one is just a tidy up of what's there (as you say it has this old code) + fix reboot issue.
commit two saves some memory by releasing the environment before TSR
commit three changes to use CARRY flag, passes error reason / max function in BX